src/compiler/parser.ts

```diff
@@ -4936,8 +4937,11 @@ namespace ts {
                 else if (lookAhead(nextTokenIsDot)) {
                     // This is an 'import.*' metaproperty (i.e. 'import.meta')
                     nextToken(); // advance past the 'import'
+                    const importEnd = scanner.getStartPos();
                     nextToken(); // advance past the dot
-                    expression = finishNode(factory.createMetaProperty(SyntaxKind.ImportKeyword, parseIdentifierName()), pos);
+                    const metaProperty = factory.createMetaProperty(SyntaxKind.ImportKeyword, parseIdentifierName());
+                    expression = finishNode(metaProperty, pos);
+                    setTextRangePosEnd(metaProperty.expression, pos, importEnd);

```

Comment:
This was done in a backwards compatible way. Is there something better that could be done or perhaps should we make a breaking change to the factory function (or add a third optional parameter for the `expression`?)